portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java

```diff
@@ -0,0 +1,155 @@
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ServingUrlOptions;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.logging.Logger;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+import com.google.gson.Gson;
+import java.util.logging.Logger;
+
+/**
+ * When the user submits the form, Blobstore processes the file upload and then forwards the request
+ * to this servlet. This servlet can then process the request using the file URL we get from
+ * Blobstore.
+ */
+@WebServlet("/my-form-handler")
+public class FormHandlerServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // Get the input from the form.
+    String hobbies = getParameter(request, "hobbies-input", "");
+    String contact= getParameter(request, "contact-input", "");
+    long timestamp = System.currentTimeMillis();
+    System.out.println(hobbies);
+
+    // Get the URL of the image that the user uploaded to Blobstore.
+    String imageUrl = getUploadedFileUrl(request, "image");
+
+
+
+    if(contact.equals("")){
+        response.setContentType("text/html");
+        response.getWriter().println("Please enter a valid name and email");
+        return;
+    }
+    
+
+    // Break the text into individual words.
+    String[] words = contact.split("\\s*,\\s*");
+
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("body",hobbies);
+    commentEntity.setProperty("name", words[0]);
+    commentEntity.setProperty("email", words[1]);
+    commentEntity.setProperty("timestamp", timestamp);
+    commentEntity.setProperty("image", imageUrl);
+   
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+
+
+    PrintWriter out = response.getWriter();
```

Comment:
How useful is this page?

When I comment on other websites like YouTube, Facebook etc, there isn't a thank you page and I don't have to click an extra button to go back.

Up to you.

Reply:
Yes, I think that sounds like a better idea